environment.rst

```diff
@@ -0,0 +1,81 @@
+
+The CS Virtual Machine
+----------------------
+
+The CS department also provides a virtual machine (VM) that approximates the software environment found in the CS machines. You can find the latest version of the CS virtual machine on `this page <https://howto.cs.uchicago.edu/vm:index>`__, which also includes instructions on how to install and run the VM.
+
+Running the VM will effectively run a full Linux operating system from inside your current operating system, and will feel as if you were logging into one of the Linux machines in the CSIL labs. However, take into account that it is possible to run the VM in "Headless Mode", which is more similar to running a Linux *server* inside your computer, which you'll then be able to access following the same steps found in the :ref:`working-remotely` page. This will allow the VM to run faster and consume fewer resources, because you'll be doing your work on your personal computer, and only accessing the VM to compile and run your code. The CS VM documentation also includes `instructions on how to set up the VM in Headless Mode <https://howto.cs.uchicago.edu/vm:headless>`__.
+
+
+Recommended code editors
+------------------------
+
+This section provides a few recommendations on code editors you may want to consider. Some of them are included in the CS machines, but you can also install them on your own computer to write your code there (and then follow the steps found in the :ref:`working-remotely` page to compile and eun your code on a CS machine). All of these tools are available on Windows, Mac, and Linux.
+
+You may find that some people are very opinionated about their choice of editor, and will sometimes `argue passionately <https://en.wikipedia.org/wiki/Editor_war>`__ about why their editor is right and yours is wrong. You should ignore these people and ultimately use whatever editor works best for *you* (and don't judge other people for their choice of editor!)
+
+.. figure:: _static/editors.jpg
+   :align: center
+   :alt: Use whatever editor works for you
+
+   *Design by Borja Sotomayor, artwork by Sarah Becan*
+
+Terminal-based editors
+~~~~~~~~~~~~~~~~~~~~~~
+
+Terminal-based editors, like `vim <https://www.vim.org/>`__ and `Emacs <https://www.gnu.org/software/emacs/>`__, have the advantage of being found on practically every UNIX system and not requiring a graphical desktop, which means they can be used when logging into a machine remotely via SSH, or on older machines that may feel sluggish when running some of the graphical editors we discuss below. They can also be extremely powerful, and can be customized to work with pretty much any programming language under the sun.
```

Comment:
The instructions for the UChicago VM setup use Nano, not sure if that is worth throwing in here as well

Reply:
Added a link to nano.